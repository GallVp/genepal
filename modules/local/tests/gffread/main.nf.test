nextflow_process {

    name "Test Process GFFREAD"
    script "../../../nf-core/gffread/main.nf"
    config "./nextflow.config"
    process "GFFREAD"

    tag "gffread"
    tag "modules_nfcore"
    tag "modules"

    test("filter by length") {

        when {
            params {
                gffread_params = '--no-pseudo --keep-genes -C -l 72'
            }
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file("$baseDir" + '/modules/local/tests/gffread/testdata/t.gff', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert file(process.out.gffread_gff[0][1]).text.contains('gene19851') },
                { assert file(process.out.gffread_gff[0][1]).text.contains('gene19851.t1') },
                { assert ! file(process.out.gffread_gff[0][1]).text.contains('gene19851.t2') } // This is the only transcript which is being knocked out
            )
        }

    }

    test("filter by length -- case II") {

        when {
            params {
                gffread_params = '--no-pseudo --keep-genes -C -l 72'
            }
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file("$baseDir" + '/modules/local/tests/gffread/testdata/t2.gff', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            with (file(process.out.gffread_gff[0][1]).text) {
                assert contains('P005ME_h2_v2.g24')

                // The following should have been excluded if CDS length was considered by gffread
                assert contains('P005ME_h2_v2.g14880')
                assert contains('P005ME_h2_v2.g19822')
                assert contains('P005ME_h2_v2.g8694')
            }
        }

    }

    test("Exclude models with in-frame stop codons") {

        setup {
            run("GUNZIP") {
                script "../../../nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = [
                        [id: 'test'],
                        file("$baseDir" + '/modules/local/tests/gffread/testdata/inframe.stop.fa.gz', checkIfExists: true)
                    ]
                    """
                }
            }
        }

        when {
            params {
                gffread_params = '-J'
            }
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file("$baseDir" + '/modules/local/tests/gffread/testdata/inframe.stop.gff', checkIfExists: true)
                ]
                input[1] = GUNZIP.out.gunzip.map { meta, fa -> fa }
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            with (file(process.out.gffread_gff[0][1]).text) {
                assert contains('g33587')

                // The following should have been excluded if in-frame stop codons were considered by gffread
                assert ! contains('g33586')
            }
        }

    }

}
