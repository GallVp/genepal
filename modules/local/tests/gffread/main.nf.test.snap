{
    "Exclude models with in-frame stop codons": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,27b4dbccace0637328733bcad6349bb8"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,05f671c6c6e530acedad0af0a5948dbd"
                ],
                "gffread_fasta": [
                    
                ],
                "gffread_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,27b4dbccace0637328733bcad6349bb8"
                    ]
                ],
                "gtf": [
                    
                ],
                "versions": [
                    "versions.yml:md5,05f671c6c6e530acedad0af0a5948dbd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-02T15:29:04.830714"
    },
    "filter by length -- case II": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,bfe869b0da10fa3a5099abacb710e205"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,05f671c6c6e530acedad0af0a5948dbd"
                ],
                "gffread_fasta": [
                    
                ],
                "gffread_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,bfe869b0da10fa3a5099abacb710e205"
                    ]
                ],
                "gtf": [
                    
                ],
                "versions": [
                    "versions.yml:md5,05f671c6c6e530acedad0af0a5948dbd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-04T14:36:20.529236"
    },
    "filter by length": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,59a7d6ff7123589ef2b90b20043a347c"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,05f671c6c6e530acedad0af0a5948dbd"
                ],
                "gffread_fasta": [
                    
                ],
                "gffread_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,59a7d6ff7123589ef2b90b20043a347c"
                    ]
                ],
                "gtf": [
                    
                ],
                "versions": [
                    "versions.yml:md5,05f671c6c6e530acedad0af0a5948dbd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-04T14:36:16.614311"
    }
}